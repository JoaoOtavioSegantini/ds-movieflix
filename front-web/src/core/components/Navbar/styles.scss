.main-nav {
  height: 70px;
  background: var(--primary);
  width: auto;
  flex: 1;
  align-items: center;
  margin-right: 0;
  margin-left: 0;

}

.navbar {
  width: 100vw;
  margin: 0;
  padding: 20px 40px;

}

.logo-text {
  color: var(--dark);
  font-weight: 700;
  font-size: 24px;
  line-height: 33px;
  letter-spacing: -0.015em;
  padding-left: 40px;
}

.btn {
  border: 1px solid var(--dark);
  width: 103px;
  height: 30px;
  border-radius: 10px;
  font-weight: bold;
  font-size: 14px;
  line-height: 19px;
  text-align: center;
  letter-spacing: -0.015em;
  padding: 4px 8px;

  align-items: center;
  justify-content: center;
}


@media (max-width: 576px) {
  .logo-text {
    font-size: 1em;
    margin-left: -15vw;
    margin-right: 6em;
  }

  .main-nav {
  width: 100vw;
  }

  .auth-button {
    width: 30vw;
  }

  .navbar {
    justify-content: space-around;
  }
}

@media (max-width: 300px) {
  .navbar {
    justify-content: space-between;
  }
  .logo-text {
    margin: 0 4px 0 -1em;
  }
}

