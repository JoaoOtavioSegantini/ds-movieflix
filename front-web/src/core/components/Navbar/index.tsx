import './styles.scss'

import { getAccessTokenDecoded, isTokenValid, logout } from '@utils/auth'
import { useEffect, useState } from 'react'
import { Link, useLocation } from 'react-router-dom'

const Navbar = () => {
  const [currentUser, setCurretUser] = useState('')
  const { user_name } = getAccessTokenDecoded()
  const isValid = isTokenValid()
  const location = useLocation()

  useEffect(() => {
    setCurretUser(user_name)
  }, [location, user_name])

  const handleLogout = (event: React.MouseEvent<HTMLDivElement>) => {
    event.preventDefault()
    logout()
  }

  return (
    <nav className="row bg-primary main-nav">
      <div className="navbar">
        <Link to="/" className="logo-text col-3">
          MovieFlix
        </Link>
        {isValid && (
          <div
            title={currentUser}
            className="col-9 auth-button btn btn-primary"
            onClick={handleLogout}
          >
            SAIR
          </div>
        )}
      </div>
    </nav>
  )
}

export default Navbar
