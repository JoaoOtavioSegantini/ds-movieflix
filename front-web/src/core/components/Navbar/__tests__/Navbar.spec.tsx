import { render, screen } from '@testing-library/react'
import Navbar from '..'
import history from 'src/core/utils/history'
import { Router } from 'react-router-dom'
import userEvent from '@testing-library/user-event'
import SessionMock from '@mock/sessionMock'

describe('<Navbar />', () => {
  test('should render <Navbar /> component', () => {
    const { container } = render(
      <Router history={history}>
        <Navbar />
      </Router>
    )
    const title = screen.getByText('MovieFlix')
    const button = screen.queryByText('SAIR')

    expect(title).toBeInTheDocument()
    expect(title).toHaveStyle({ backgroundColor: 'var(--primary)' })
    expect(button).not.toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
  test('should rende button when user is logged', () => {
    new SessionMock().userValidMember()

    const { container } = render(
      <Router history={history}>
        <Navbar />
      </Router>
    )

    const user = screen.getByTitle('alex@gmail.com')
    const button = screen.getByText('SAIR')

    expect(button).toBeInTheDocument()
    expect(user).toBeInTheDocument()
    expect(container.firstChild).toMatchInlineSnapshot(`
      <nav
        class="row bg-primary main-nav"
      >
        <div
          class="navbar"
        >
          <a
            class="logo-text col-3"
            href="/"
          >
            MovieFlix
          </a>
          <div
            class="col-9 auth-button btn btn-primary"
            title="alex@gmail.com"
          >
            SAIR
          </div>
        </div>
      </nav>
    `)

    userEvent.click(button)
    // expect(handleLogout.mock.calls.length).toEqual(1)
    expect(button.onclick?.call.length).toEqual(1)
  })
})
