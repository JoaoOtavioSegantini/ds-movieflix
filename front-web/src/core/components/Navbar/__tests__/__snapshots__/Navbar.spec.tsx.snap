// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render <Navbar /> component 1`] = `
<div>
  <nav
    class="row bg-primary main-nav"
  >
    <div
      class="navbar"
    >
      <a
        class="logo-text col-3"
        href="/"
      >
        MovieFlix
      </a>
    </div>
  </nav>
</div>
`;
